src/azul/drs.py

```diff
@@ -74,10 +87,133 @@ class Access:
     headers: Optional[Mapping[str, str]] = None
 
 
+class DRSURI(metaclass=ABCMeta):
+    scheme: str
+    netloc: str
+    object_id: str
+
+    @classmethod
+    def parse(cls, drs_uri: str) -> 'DRSURI':
+        try:
+            drs_uri = RegularDRSURI.parse(drs_uri)
+        except ValueError as e:
+            if len(e.args) == 1 and isinstance(e.args[0], str) and e.args[0].startswith('Invalid port '):
+                drs_uri = CompactDRSURI.parse(drs_uri)
+            else:
+                raise
+        return drs_uri
+
+    @abstractmethod
+    def expand(self, client: 'DRSClient') -> mutable_furl:
```

Comment:
```suggestion
    def to_url(self, client: 'DRSClient') -> mutable_furl:
```

And docstring, please.